### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by gumaerc**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/18

#### What's this PR do?
Adds unit tests to assert output of each function

#### How should this be manually tested?
Clone the repo, run `yarn install`, then `npm test`

### CODE REVIEW

File: src/file_operations_test.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it("throws an error when you call it with no source directory", () => {
+    try {
+      fileOperations.scanCourses(null, destinationPath)
+    } catch (err) {
+      assert.equal(err.message, "Invalid source directory")
+    }
```
Comment (aliceriot) [inline src/file_operations_test.js]: you might be able to use the chai `assert.throws` method for these, although what you have here works too

File: src/file_operations_test.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      assert.equal(err.message, "Invalid destination directory")
+    }
+  })
+
+  it("calls readdirSync once", () => {
+    fileOperations.scanCourses(sourcePath, destinationPath)
+    assert(
+      readdirSync.calledOnce,
+      "Expected readdirSync to be called once on the source path"
+    )
```
Comment (aliceriot) [inline src/file_operations_test.js]: it's not crucial, but `sinon` has some assertion methods that will produce nicer error messages when they fail (iirc it's like `sinon.assert.calledOnce(stub)`)

File: src/file_operations_test.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  })
+
+  afterEach(() => {
+    sandbox.restore()
+  })
+
+  it("calls mkDirSync to create sections folder", () => {
+    mkDirSync.calledWithExactly(
+      path.join(destinationPath, singleCourseId, "sections")
+    )
```
Comment (aliceriot) [inline src/file_operations_test.js]: I think you forgot the assert on this one

File: src/markdown_generators_test.js [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    )
+    const expectedSections = singleCourseJsonData["course_pages"]
+      .filter(page => page["text"])
+      .map(page => page["short_url"])
+    const sections = singleCourseMarkdownData.map(section => {
+      return section["name"]
+    })
+    assert(
+      sections.includes("_index.md"),
+      "expected _index.md to be in the markdown data"
```
Comment (aliceriot) [inline src/markdown_generators_test.js]: theres an `assert.include` you may be able to use here

File: src/markdown_generators_test.js [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+    sandbox.restore()
+  })
+
+  it(`sets the title of the page to "Course Home"`, () => {
+    const expectedValue = "Course Home"
+    const foundValue = courseHomeFrontMatter["title"]
+    assert.equal(
+      expectedValue,
+      foundValue,
+      `expected ${expectedValue} to equal ${foundValue}`
```
Comment (aliceriot) [inline src/markdown_generators_test.js]: I think if you use `assert.equal` the message that it prints will be pretty much what you've provided here by default

File: src/file_operations.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+const readdir = util.promisify(fs.readdir)
+let directoriesScanned = 0
+
+const directoryExists = directory => {
+  return (
+    directory &&
+    fs.existsSync(directory) &&
+    fs.lstatSync(directory).isDirectory()
+  )
+}
```
[the remark this answers is not in the document]
Comment (gumaerc) [reply src/file_operations.js]: Seriously... it should at least be a part of `fs`

File: package.json [hunk 7]
```diff
@@ -47,6 +47,7 @@
     "prettier-eslint": "^9.0.1",
     "prettier-eslint-cli": "^5.0.0",
     "sinon": "^8.0.4",
+    "sinon-chai": "^3.4.0",
```
Comment (aliceriot) [inline package.json]: ooh, I don't think we're using this elsewhere, might have to add that to open soon...

**Pull request merged**